dataset.go

```diff
@@ -73,6 +89,18 @@ func (d *Dataset) FlatIterator() <-chan *Element {
 	return elemChan
 }
 
+// ExhaustElementChannel exhausts the channel iterator returned by
+// Dataset.FlatIterator, ensuring that the underlying Go routine completes.
+// When using Dataset.FlatIterator, if your program will exit for some reason
+// without reading all the elements of the channel, you should be sure to call
+// this function to prevent a phantom Goroutine.
+// Or, if you don't need the channel interface, simply use
+// Dataset.FlatStatefulIterator.
+func ExhaustElementChannel(c <-chan *Element) {
```

Comment:
Maybe it would make sense to make this a method on the retuned channel using a custom type? 

Something like this:

```go
// FlatIterChan is returned from Dataset.FlatIterator() and carries seme QoL methods.
type FlatIterChan <-chan *Element

// ExhaustElementChannel exhausts the channel iterator returned by
// Dataset.FlatIterator, ensuring that the underlying Go routine completes.
// When using Dataset.FlatIterator, if your program will exit for some reason
// without reading all the elements of the channel, you should be sure to call
// this function to prevent a phantom Goroutine.
// Or, if you don't need the channel interface, simply use
// Dataset.FlatStatefulIterator.
func (iterChan FlatIterChan) ExhaustElements(c <-chan *Element) {
	for _ = range c {
	}
}

// FlatIterator returns a channel upon which every element in this Dataset will be sent,
// including elements nested inside sequences.
// Note that the sequence element itself is sent on the channel in addition to the child elements in the sequence.
// TODO(suyashkumar): decide if the sequence element itself should be sent or not
func (d *Dataset) FlatIterator() FlatIterChan {
	elemChan := make(chan *Element)
	go func() {
		flatElementsIterator(d.Elements, elemChan)
		close(elemChan)
	}()
	return elemChan
}
```

Any caller methods that take `<-chan *dicom.Element` should still work so this should not be a breaking change, and feels a little more intuitive to me, since this function only really exists to be called on the returned channel from that particular method.

Reply:
+1 I like this! I didn't really think to use a type alias for channel types in this way, but I think it organizes the code better. Thanks for the suggestion!

Reply:
So, I think that type definitions like this would indeed break code that takes a `<-chan *Element`, since it's a new distinct type (though folks could cast back to a `<-chan *Element`). We could try a type alias but need to see how defining functions on a type alias works. My guess is that most consumers just range over the channel though, and they would have to modify their code anyway to add in the ExhaustElements call.

Reply:
So, for the reason above I'm going to leave it as is (and also, this API will be phased out anyway as we discussed). 

This way, the API did not break at all, and the code change needed to add ExhaustElements is the same amount of work either way (and very similar in form).